.github/workflows/regen-yarn-lock-and-copy.yml

```diff
@@ -0,0 +1,43 @@
+#
+# Copyright (c) 2023 Red Hat, Inc.
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+
+# every time root/yarn.lock is updated, copy it to tools/build/yarn.lock 
+# so that downstream builds can use the second yarn.lock for cachito processes
+
+# Note: to regenerate the yarn.lock file, use:
+# rm -f yarn.lock tools/build/yarn.lock; yarn install --mode update-lockfile; cp yarn.lock tools/build/yarn.lock
+name: Copy yarn.lock to tools/build/
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - yarn.lock 
+
+jobs:
+  build:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+          token: ${{secrets.CHE_BOT_GITHUB_TOKEN}}
+      - name: Copy yarn.lock
+        run: |
+          cp -f yarn.lock tools/build/yarn.lock
```

Comment:
probably we have to check if tools/build/yarn.lock exists and remove it before copying